src/authStrategies/RemoteAuth.js

```diff
@@ -0,0 +1,221 @@
+    constructor({ clientId, dataPath, store, backupSyncMs } = {}) {
+        super();
+
+        const idRegex = /^[-_\w]+$/i;
+        if (clientId && !idRegex.test(clientId)) {
+            throw new Error('Invalid clientId. Only alphanumeric characters, underscores and hyphens are allowed.');
+        }
+        if (!backupSyncMs || backupSyncMs < 60000) {
+            throw new Error('Invalid backupSyncMs. Accepts values starting from 60000ms {1 minute}.');
+        }
+        if(!store) throw new Error('Remote database store is required.');
+
+        this.store = store;
+        this.clientId = clientId;
+        this.backupSyncMs = backupSyncMs;
+        this.dataPath = path.resolve(dataPath || './.wwebjs_auth/');
+        this.requiredDirs = ['Default', 'IndexedDB', 'Local Storage']; /* => Required Files & Dirs in WWebJS to restore session */
+    }
+
+    async beforeBrowserInitialized() {
+        const puppeteerOpts = this.client.options.puppeteer;
+        const sessionDirName = this.clientId ? `session-${this.clientId}` : 'session';
+        const dirPath = path.join(this.dataPath, sessionDirName);
+
+        if (puppeteerOpts.userDataDir && puppeteerOpts.userDataDir !== dirPath) {
+            throw new Error('RemoteAuth is not compatible with a user-supplied userDataDir.');
+        }
+
+        this.userDataDir = dirPath;
+        this.sessionName = sessionDirName;
+
+        await this.extractRemoteSession();
+
+        this.client.options.puppeteer = {
+            ...puppeteerOpts,
+            userDataDir: dirPath
+        };
+    }
+
+    async logout() {
+        await this.deleteRemoteSession();
+
+        let pathExists = await this.isValidPath(this.userDataDir);
+        if (pathExists) {
+            await fs.promises.rm(this.userDataDir, {
+                recursive: true,
+                force: true
+            }).catch(() => {});
+        }
+        clearInterval(this.backupSync);
+    }
+
+    async authReady() {
+        const sessionExists = await this.store.sessionExists({session: this.sessionName});
+        if(!sessionExists) {
+            await this.delay(30000); /* Initial delay sync required for session to be stable enough to recover */
+            await this.storeRemoteSession();
+        }
+        var self = this;
+        this.backupSync = setInterval(async function () {
+            await self.storeRemoteSession();
+        }, this.backupSyncMs);
+    }
+
+    async storeRemoteSession() {
+        try {
+            const sessionExists = await this.store.sessionExists({session: this.sessionName});
+            if (sessionExists) {
+                await this.store.delete({session: this.sessionName});
+            }
+            /* Compress & Store Session */
+            const pathExists = await this.isValidPath(this.userDataDir);
+            if (pathExists) {
+                await this.compressSession();
+                await this.store.save({session: this.sessionName});
```

Comment:
If the interface between the authstrategy and the store will be a file, we should be more explicit about where that file is located. I would suggest passing a filepath directly instead of determining it implicitly from within the store

Reply:
> Done